pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/internal/AnnotationSuppressionUtil.java

```diff
@@ -107,9 +106,8 @@ private static boolean hasSuppressWarningsAnnotationFor(final Annotatable node,
 
     // @formatter:on
     private static boolean annotationSuppresses(ASTAnnotation annotation, Rule rule) {
-        // if (SuppressWarnings.class.equals(getType())) { // typeres is not always on
-        if (TypeHelper.isA(annotation, SuppressWarnings.class)) {
-            for (ASTLiteral element : annotation.findDescendantsOfType(ASTLiteral.class)) {
+        if (annotation.getSymbol().getBinaryName().equals("java.lang.SuppressWarnings")) {
```

Comment:
We'll get a PositionLiteralFirstInComparison violation

```suggestion
        if ("java.lang.SuppressWarnings".equals(annotation.getSymbol().getBinaryName())) {
```